pkg/tagger/collectors/docker_extract_test.go

```diff
@@ -0,0 +1,130 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+// +build docker
+
+package collectors
+
+import (
+	"testing"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDockerRecordsFromInspect(t *testing.T) {
+	testCases := []struct {
+		co                   *types.ContainerJSON
+		toRecordEnvAsTags    map[string]string
+		toRecordLabelsAsTags map[string]string
+		expectedLow          []string
+		expectedHigh         []string
+	}{
+		{
+			co: &types.ContainerJSON{
+				Config: &container.Config{
+					Env:    []string{"k=v"},
+					Labels: map[string]string{"labelKey": "labelValue"},
+				},
+			},
+			toRecordEnvAsTags:    map[string]string{},
+			toRecordLabelsAsTags: map[string]string{},
+			expectedLow:          []string{},
+			expectedHigh:         []string{},
+		},
+		{
+			co: &types.ContainerJSON{
+				Config: &container.Config{
+					Env:    []string{"k=v"},
+					Labels: map[string]string{"labelKey": "labelKey"},
+				},
+			},
+			toRecordEnvAsTags:    map[string]string{"k": "becomeK"},
+			toRecordLabelsAsTags: map[string]string{"labelKey": "labelValue"},
+			expectedLow:          []string{"becomeK:v"},
+			expectedHigh:         []string{},
+		},
+		{
+			co: &types.ContainerJSON{
+				Config: &container.Config{
+					Env:    []string{"k=v"},
+					Labels: map[string]string{"labelKey": "labelKey"},
+				},
+			},
+			toRecordEnvAsTags:    map[string]string{"k": "+becomeK"},
+			toRecordLabelsAsTags: map[string]string{"labelKey": "labelValue"},
+			expectedLow:          []string{},
+			expectedHigh:         []string{"becomeK:v"},
+		},
+		{
+			co: &types.ContainerJSON{
+				Config: &container.Config{
+					Env:    []string{"k=v", "l=t"},
+					Labels: map[string]string{"labelKey": "labelValue"},
+				},
+			},
+			toRecordEnvAsTags:    map[string]string{"k": "+becomeK", "l": "expectedLow"},
+			toRecordLabelsAsTags: map[string]string{"labelkey": "labelKey"},
+			expectedLow:          []string{"expectedLow:t", "labelKey:labelValue"},
+			expectedHigh:         []string{"becomeK:v"},
+		},
+		{
+			co: &types.ContainerJSON{
+				Config: &container.Config{
+					Env:    []string{"k=v", "l=t"},
+					Labels: map[string]string{"labelKey": "labelValue"},
+				},
+			},
+			toRecordEnvAsTags:    map[string]string{"k": "+becomeK", "l": "expectedLow"},
+			toRecordLabelsAsTags: map[string]string{"labelkey": "+labelKey"},
+			expectedLow:          []string{"expectedLow:t"},
+			expectedHigh:         []string{"becomeK:v", "labelKey:labelValue"},
+		},
+		{
+			co: &types.ContainerJSON{
+				Config: &container.Config{
+					Env: []string{
+						"MARATHON_APP_ID=1",
```

Comment:
Because our code might break because the orchestrator's output changes from version to version, we use test cases extracted from real use cases instead of placeholder values. Please reuse the test values from https://github.com/DataDog/dd-agent/blob/master/tests/core/test_mesosutil.py

This will allow us to see with what format we coded and tested the extraction logic, and pinpoint faster why it breaks.